appserver/web/web-ha/pom.xml

```diff
@@ -69,16 +69,28 @@
             <groupId>org.glassfish.main.web</groupId>
             <artifactId>web-glue</artifactId>
             <version>${project.version}</version>
+            <exclusions>
```

Comment:
This is one is wrong.
We shouldn't exclude dependencies on GF's own modules. If the dependency graph for web-glue is bad, fix it on web-glue, not in modules depending on web-glue.

> I see the ejb-ejb31-singleton-hello-ejb test failed.
> Can the test fail intermittently? It failed for me locally when running all the tests, but then after re-run the test passed.

I've seen the EJB test fail intermittently in the pipeline.

Reply:
Web-glue has it right, but when it comes to web-ha, it is not. That's the thing

Reply:
Web-glue has "what" done right ? Are you talking about the manifest or the dependency tree ?
If the latter then obviously it's not right since you have to put some excludes.

Reply:
web-glue has the depdendency on GF 5.0.1. web-ha has then dependency on 5.1.0. 
[INFO] org.glassfish.main.web:web-ha:glassfish-jar:5.0.1-SNAPSHOT
[INFO] +- org.glassfish.main.web:web-glue:jar:5.0.1-SNAPSHOT:compile
[INFO] |  +- javax.servlet:javax.servlet-api:jar:4.0.0:compile
[INFO] |  +- org.glassfish.main.web:web-cli:jar:5.0.1-SNAPSHOT:compile
[INFO] |  +- org.glassfish.main.flashlight:flashlight-framework:jar:5.0.1-SNAPSHOT:compile
[INFO] |  +- org.glassfish.hk2:hk2-core:jar:2.5.0:compile
[INFO] |  +- org.glassfish.main.common:stats77:jar:5.0.1-SNAPSHOT:compile
[INFO] |  +- org.glassfish.main.web:war-util:jar:5.0.1-SNAPSHOT:compile
[INFO] |  +- org.glassfish.main.web:web-naming:jar:5.0.1-SNAPSHOT:compile
[INFO] |  +- org.glassfish.main.web:web-core:jar:5.0.1-SNAPSHOT:compile
[INFO] |  +- org.glassfish.main.admin:admin-util:jar:5.0.1-SNAPSHOT:compile
[INFO] |  +- org.glassfish.main.deployment:deployment-common:jar:5.0.1-SNAPSHOT:compile
[INFO] |  +- org.glassfish.main.deployment:deployment-javaee-core:jar:5.0.1-SNAPSHOT:compile
[INFO] |  +- org.glassfish.web:javax.servlet.jsp:jar:2.3.4:compile
[INFO] |  +- org.glassfish:javax.el:jar:3.0.1-b11:compile
[INFO] |  +- org.glassfish.main.admin:admin-core:jar:5.0.1-SNAPSHOT:compile
[INFO] |  +- org.glassfish.main.admin:monitoring-core:jar:5.0.1-SNAPSHOT:compile
[INFO] |  +- org.glassfish.main.deployment:dol:jar:5.0.1-SNAPSHOT:compile
[INFO] |  +- org.glassfish.main.common:glassfish-api:jar:5.0.1-SNAPSHOT:compile
[INFO] |  +- org.glassfish.main.common:internal-api:jar:5.0.1-SNAPSHOT:compile
[INFO] |  +- org.glassfish.main.transaction:transaction-internal-api:jar:5.0.1-SNAPSHOT:compile
[INFO] |  +- org.glassfish.main.common:container-common:jar:5.1.0:compile

Reply:
[INFO] org.glassfish.main.web:web-glue:glassfish-jar:5.0.1-SNAPSHOT
[INFO] +- jakarta.servlet:jakarta.servlet-api:jar:4.0.2:compile
[INFO] +- org.glassfish.main.web:web-cli:jar:5.0.1-SNAPSHOT:compile
[INFO] +- org.glassfish.main.flashlight:flashlight-framework:jar:5.0.1-SNAPSHOT:compile
[INFO] +- org.glassfish.hk2:hk2-core:jar:2.5.0:compile
[INFO] +- org.glassfish.main.common:stats77:jar:5.0.1-SNAPSHOT:compile
[INFO] +- org.glassfish.main.web:war-util:jar:5.0.1-SNAPSHOT:compile
[INFO] +- org.glassfish.main.web:web-naming:jar:5.0.1-SNAPSHOT:compile
[INFO] +- org.glassfish.main.web:web-core:jar:5.0.1-SNAPSHOT:compile
[INFO] +- org.glassfish.main.admin:admin-util:jar:5.0.1-SNAPSHOT:compile
[INFO] +- org.glassfish.main.common:common-util:jar:5.0.1-SNAPSHOT:compile
[INFO] +- org.glassfish.main.deployment:deployment-common:jar:5.0.1-SNAPSHOT:compile
[INFO] +- org.glassfish.main.deployment:deployment-javaee-core:jar:5.0.1-SNAPSHOT:compile
[INFO] +- org.glassfish.web:javax.servlet.jsp:jar:2.3.4:compile
[INFO] +- org.glassfish:jakarta.el:jar:3.0.2:compile
[INFO] +- org.glassfish.main.admin:admin-core:jar:5.0.1-SNAPSHOT:compile
[INFO] +- org.glassfish.main.admin:monitoring-core:jar:5.0.1-SNAPSHOT:compile
[INFO] +- org.glassfish.main.deployment:dol:jar:5.0.1-SNAPSHOT:compile
[INFO] +- org.glassfish.main.common:glassfish-api:jar:5.0.1-SNAPSHOT:compile
[INFO] +- org.glassfish.main.common:internal-api:jar:5.0.1-SNAPSHOT:compile
[INFO] +- org.glassfish.main.transaction:transaction-internal-api:jar:5.0.1-SNAPSHOT:compile
[INFO] +- org.glassfish.main.common:container-common:jar:5.0.1-SNAPSHOT:compile

Reply:
hmm, interesting...
Does web-ha has a compile-time dependency on container-common ?

Reply:
? [INFO] | +- org.glassfish.main.common:container-common:jar:5.1.0:compile

Reply:
This is a dependency is "compile" scope resolve transitively via web-glue. 
I'm asking if web-ha has code compiled against container-common. If this is the case you can declare a direct dependency on container-common with version=project.version.

Reply:
So if it has, I can add the direct dependency, which is what I am doing. If has not, then I can exclude?

Reply:
If it didn't have a compile dependency then I would first want to understand this maven graph mismatch. exclude should used only as a last resort in case we can't explain that and we have not proper fix.

What puzzles me is that you said that the dependency on 5.1.0 has been removed from jersey so why are you still seeing these ?

Reply:
Make sure your local environment is clean. You can re-build both jersey and glassfish locally with a fresh local repository.